service/history/workflowExecutionContext.go

```diff
@@ -168,8 +172,19 @@ func (c *workflowExecutionContext) updateWorkflowExecution(transferTasks []persi
 	// Only generate replication task if this is a global domain
 	createReplicationTask := c.msBuilder.replicationState != nil
 
-	lastWriteVersion := c.msBuilder.GetCurrentVersion()
-	return c.updateHelper(nil, transferTasks, timerTasks, createReplicationTask, "", lastWriteVersion, transactionID)
+	currentVersion := c.msBuilder.GetCurrentVersion()
+	if createReplicationTask {
+		lastWriteVersion := c.msBuilder.GetLastWriteVersion()
+		activeCluster := c.clusterMetadata.ClusterNameForFailoverVersion(currentVersion)
+		currentCluster := c.clusterMetadata.GetCurrentClusterName()
+		if currentVersion < lastWriteVersion {
```

Comment:
I think we should make sure to now let CurrentVersion go back to previous value once it detects a higher version.
Once we have this guarantee, then this check is no longer needed.